### ISSUE TITLE
Do not set Fields "Touched" that just get Activated and Blurred without Changes

### ISSUE THREAD

**Issue by fabb** [jaredpalmer/formik#1364]
### Current Behavior

When a field is empty, and the user clicks into the field and then blurs it again, the field is set to `touched`, and validation is displayed.

### Desired Behavior

It would be nice if the field would not get set `touched` in that case, and validation not appear, since some users click through a lot of fields in the beginning and this feels not so nice.

### Suggested Solution

* In `handleBlur` do not set the field to `touched` in case the field is empty.
* Maybe add a configuration option to make this optional for library clients.
* Maybe do set `touched` for fields that have been non-empty initially, and are blurred.

### Who does this impact? Who is this for?

This is for UX for end users.

### Describe alternatives you've considered

Wrap wrap `handleBlur` with a method inside of the form, but that is inconvenient and would need to be done for every form and every field.

### Additional context

This is the piece of code that needs to be changed: https://github.com/jaredpalmer/formik/blob/master/src/Formik.tsx#L482

**Comment by knodel12** [comment]
I wonder if having a different field would be better for this... Instead of touched, changed?

**Comment by fabb** [comment]
You mean because I did physically touch the field? I don‘t know, imho it fits quite well in the current semantics of touched and changed as a different working mode.

**Comment by jhoch** [comment]
Seems similar to ([`dirty`](https://jaredpalmer.com/formik/docs/api/formik#dirty-boolean)), but for individual fields, as opposed to the form in its entirety.

**Comment by hegelstad** [comment]
Its possible to pass dirty-render prop from Formik down to the ErrorMessage component and render the ErrorMessage component only when the form is dirty. That would solve it partially, but not on a individual field level.

**Comment by jhoch** [comment]
You can write your own logic using `FieldProps`:
```typescript
(props: FieldProps<V>) => {
      const { field: { name, value }, form } = props;
      const initialValue = getIn(form.initialValues, name);
      const className = isEqual(value, initialValue) ? 'clean' : 'dirty';
}
```
and bake this into your component (I followed a similar approach recently).  Curious if others see a downside here.

I could see it being expensive on large forms, so it wouldn't want to be done by default in Formik on all forms.

**Comment by jaredpalmer** [comment]
it is pretty easy to calculate as @jhoch pointed out. I would call what this is describing as another name, perhaps `visited` vs. `touched` vs. `modified` would be clearer

**Comment by r00dY** [comment]
I'm 100% for this feature. In my opinion current `touched` flag promotes bad UX. Let me explain why I think so.

If you focus input and then focus out without changing it, what's the possible scenario why you did it?
1. You're filling the form going through it using tab and want to leave field empty (validation is ok here)
2. You're just passing through the fields with a tab to get to some other place of the website (validation not ok).
3. You focused the field unintentionally and just continue using website (validation not ok).
4. You clicked the field because... you like to click. Some people just click around (validation not ok).
5. You wanted to fill the input but changed your mind and do not want to fill the form at all (validation not necessary).

In summary, validation very frequently doesn't make sense. But also, very often it does. We can't decide.

And most important thing is that

if I can't decide, I do not want to show validation. Validation is intrusive. Error messages are not nice. People might get negative emotions looking at red error messages. It looks as if they did sth wrong even if they didn't. So if you don't know, don't risk making them feel bad.

The form will be validated anyway on submit. If the field is required, users will fill it.

Curious if you agree.
